src/Std/Data/DHashMap/Lemmas.lean

```diff
@@ -964,4 +965,701 @@ theorem distinct_keys [EquivBEq α] [LawfulHashable α] :
+  sorry
+
+@[simp]
+theorem mem_modify [LawfulBEq α] {k k': α} {f : β k → β k} : k' ∈ m.modify k f ↔ k' ∈ m := by
+  simp only [mem_iff_contains, contains_modify]
+
+@[simp]
+theorem size_modify [LawfulBEq α] {k : α} {f : β k → β k} : (m.modify k f).size = m.size :=
+  sorry
+
+theorem get?_modify [LawfulBEq α] {k k' : α} {f : β k → β k} :
+    (m.modify k f).get? k' = if h : k == k' then
+      (cast (congrArg (Option ∘ β) (eq_of_beq h)) ((m.get? k).map f))
+    else
+      m.get? k' :=
+  sorry
+
+@[simp]
+theorem get?_modify_self [LawfulBEq α] {k : α} {f : β k → β k} :
+    (m.modify k f).get? k = (m.get? k).map f :=
+  sorry
+
+theorem get_modify [LawfulBEq α] {k k' : α} {f : β k → β k}
+    (h : k' ∈ m.modify k f) :
+    (m.modify k f).get k' h =
+    if heq : k == k' then
+      haveI h' : k ∈ m := by rwa [mem_modify, ← eq_of_beq heq] at h
+      cast (congrArg β (eq_of_beq heq)) <| f (m.get k h')
+    else
+      haveI h' : k' ∈ m := by rwa [mem_modify] at h
+      m.get k' h' :=
+  sorry
+
+@[simp]
+theorem get_modify_self [LawfulBEq α] {k : α} {f : β k → β k} {h : k ∈ m.modify k f} :
+    haveI h' : k ∈ m := by rwa [mem_modify] at h
+    (m.modify k f).get k h = f (m.get k h') :=
+  sorry
+
+theorem get!_modify [LawfulBEq α] {k k' : α} [hi : Inhabited (β k')] {f : β k → β k} :
+    (m.modify k f).get! k' =
+    if heq : k == k' then
+      haveI : Inhabited (β k) := ⟨cast (congrArg β <| eq_of_beq heq).symm default⟩
+      -- not correct if f does not preserve default: ... f (m.get! k)
+      -- possible alternative: write ... (m.getD k (cast ⋯ default))
+      m.get? k |>.map f |>.map (cast (congrArg β (eq_of_beq heq))) |>.get!
+    else
+      m.get! k' :=
+  sorry
+
+@[simp]
+theorem get!_modify_self [LawfulBEq α] {k : α} [Inhabited (β k)] {f : β k → β k} :
+    (m.modify k f).get! k = ((m.get? k).map f).get! :=
+  sorry
+
+theorem getD_modify [LawfulBEq α] {k k' : α} {f : β k → β k} :
+    (m.modify k f).getD k' =
+    if heq : k == k' then
+      m.get? k |>.map f |>.map (cast (congrArg β <| eq_of_beq heq)) |>.getD
+    else
+      m.getD k' :=
+  sorry
+
+@[simp]
+theorem getD_modify_self [LawfulBEq α] {k k' : α} {f : β k → β k} :
+    (m.modify k f).getD k = ((m.get? k).map f).getD :=
+  sorry
+
+theorem getKey?_modify [LawfulBEq α] {k k' : α} {f : β k → β k} :
+    (m.modify k f).getKey? k' =
+    if k == k' then
+      Option.guard (fun _ => k ∈ m) k
+    else
+      m.getKey? k' :=
+  sorry
+
+@[simp]
+theorem getKey?_modify_self [LawfulBEq α] {k : α} {f : β k → β k} :
+    (m.modify k f).getKey? k = Option.guard (fun _ => k ∈ m) k :=
+  sorry
+
+theorem getKey!_modify [LawfulBEq α] [Inhabited α] {k k' : α} {f : β k → β k} :
+    (m.modify k f).getKey! k' =
+    if k == k' then
+      -- possible alternative (here and at other places): if k ∈ m then k else default
+      -- although correct, I'm not sure about the tradeoffs of replacing get! with
+      -- something that does not panic
```

Comment:
In my opinion, it is not the task of verification theorems to preserve panic behaviour. (Particularly since, if you forget to set the flag, panics don't actually panic...)